### ISSUE TITLE
[FEATURE]: Add support for lookup_srv for upstreams discovery

### ISSUE THREAD

**Issue by danlsgiga** [mholt/caddy-l4#15]
The same way as the caddy `http` app, it would be nice to have not only the `dial` option but also the `lookup_srv` for upstream service discovery.

**Comment by pwFoo** [comment]
I like https://github.com/lucaslorentz/caddy-docker-proxy label configuration. Something similiar would be awesome

**Comment by francislavoie** [comment]
Docker labels are out of scope for this plugin.

**Comment by pwFoo** [comment]
So not label based or entire feature of auto generated config by other containers labels / env?

**Comment by francislavoie** [comment]
caddy-l4 has nothing to do with Docker. It's not in scope for this plugin.

**Comment by johnalotoski** [comment]
It would be great to see SRV support for dynamic upstreams here.  Awesome that caddy supports SRV based lookups on the http side already.  I'm trying to wrap my head around why so few proxy products support SRV lookups.  AFAICT nginx (non-commercial) doesn't, traefik doesn't (equivalent functionality needs to use a DCS, like Consul, for service discovery), envoy doesn't even seem to yet.

Since the http app already has the SRV lookup feature (dynamic srv stanza in reverse_proxy for example), maybe the level of effort wouldn't be huge to re-use a good part of that for the tcp app?

**Comment by mholt** [comment]
@johnalotoski Yeah, this should be pretty straightforward to port over from the http app now.

**Comment by unixfox** [comment]
I know it's not great to "vibe code" on an open source project. But I asked copilot to develop this feature, and it succeeded to do so: https://github.com/unixfox/caddy-l4/pull/2. The code works I tested it with my own SRV records on consul.

If the code is somewhat "ok" I can create a PR and iterate based on the feedback (though I'm not that well experienced with Golang).

If you want to test it, you can compile caddy using:
```
xcaddy build --with github.com/mholt/caddy-l4=github.com/unixfox/caddy-l4@latest
```

**Comment by mholt** [comment]
@unixfox Eh, it's allowed as long as you verify everything that is generated, and test it, etc etc. (Still on the fence about whether "vibe coded" solutions are compatible with the CLA which asserts you have the rights to contribute the code, since it's still unclear if LLMs steal work. But that's another matter I'll let others figure out.)

Feel free to submit a PR for review! Make sure to let us know you used an LLM for assistance.
